Add region event handling module

RobertHodan opened the pull request.

Commit:
Add event region handling module

Commit:
Test region handling with demo code

Commit:
Make preventDefault optional

danielweck commented on src/modules/regionHandling/handler.ts:
```diff
@@ -0,0 +1,55 @@
+        return;
+      }
+
+      let x = (<MouseEvent> event).clientX;
```
Just a couple of thoughts:
1) shouldn't there be a check for `options.eventType` to ensure `event` is actually a `MouseEvent`? (the `eventType` is specified by the client of the API, when invoking `this.regiGlue.addEventListener()`)
2) what about unifying Pointer Events? (touch handling)

Commit:
Improve region handling

- Allow offsets to be set
- Allow options to be modified
- Return event data to callback

Commit:
Remove offset from RegionHandling module

- Rename SetOptions to SetRegion

Commit:
Fix miscalculation with new regions

Commit:
Add text selector to range data

Commit:
Add generateCFI module

Commit:
Allow highlighting using CFI's

Commit:
Iterate through all child divs

Commit:
Add readium-cfi-js dependency for modules
